### ISSUE TITLE
Faulty floating point exception due to SLP optimization

### ISSUE THREAD

**Issue by sponce** [llvm/llvm-project#77492]
The failing code is basically doing 3 similar operations which are done in one go thanks to SLP, except the 4th item in the vector used runs a 0/0 division leading to a floating point exception raised (I've enabled them).

A full reproducer can be found on godbolt : https://godbolt.org/z/sePnqh9ss
It also shows how gcc avoids the problem with to an extra instruction `movlhps` which makes sure that 0/1 is executed rather then 0/0

**Comment by topperc** [comment]
LLVM strongly assumes the default FP environment unless you add `#pragma STDC FENV_ACCESS`, `-ftrapping-math`, `-ffp-model=strict`, etc.

**Comment by phoebewang** [comment]
Agreed with @topperc . In the default FP model, the codegen doesn't promise exception correctness. You can try the way Craig provided in this case, though there're much room for us to improvide the performance under strict FP model.
